modules/flowable-engine/src/main/java/org/flowable/engine/impl/bpmn/behavior/UserTaskActivityBehavior.java

```diff
@@ -251,10 +254,16 @@ protected void handleDueDate(CreateUserTaskBeforeContext beforeContext, Expressi
                     BusinessCalendar businessCalendar = processEngineConfiguration.getBusinessCalendarManager()
                             .getBusinessCalendar(businessCalendarName);
                     task.setDueDate(businessCalendar.resolveDuedate((String) dueDate));
-
-                } else {
-                    throw new FlowableIllegalArgumentException("Due date expression does not resolve to a Date or Date string: " + activeTaskDueDate);
+                } else if (dueDate instanceof Instant) {
+                    task.setDueDate(Date.from((Instant) dueDate));
+                } else if (dueDate instanceof LocalDate) {
+                    Date localDueDate = Date.from(((LocalDate) dueDate).atStartOfDay()
+                            .atZone(ZoneId.systemDefault())
+                            .toInstant());
+                    task.setDueDate(localDueDate);
                 }
+            } else {
```

Comment:
This `else` part must be moved in the inner if-statement

Reply:
Thanks, should have caught that. I have added an additional test to test for invalid dates as well.